opentracing-spring-web-handler-interceptor-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/TracingAutoconfiguration.java

```diff
@@ -0,0 +1,66 @@
+package io.opentracing.contrib.spring.web;
+
+import java.util.Arrays;
+import java.util.logging.Logger;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureOrder;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.web.servlet.FilterRegistrationBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.Ordered;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
+
+import io.opentracing.NoopTracer;
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.web.interceptor.SpanDecorator;
+import io.opentracing.contrib.spring.web.interceptor.TracingHandlerInterceptor;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+
+/**
+ * @author Pavol Loffay
+ */
+@Configuration
+@AutoConfigureOrder(Ordered.HIGHEST_PRECEDENCE)
```

Comment:
This is rude :) I don't know Spring that well, but if it does what I think it does, do you really need this running before everything else? Usually, these levels are reserved for security-related components. We could argue that "monitoring" is at the same level as "security", though.